src/gui/ScreenHandler.cpp

```diff
@@ -164,6 +164,20 @@ void Screens::PushBeforeCurrent(const ScreenFactory::Creator screen_creator) {
 }
 
 void Screens::Loop() {
+    if (GuiDefaults::menu_timeout_enabled) {
+        window_frame_t *w = current.get();
+        if (w->IsTimeout()) {
+            gui_timeout_id = gui_get_menu_timeout_id();
+            if (gui_timer_expired(gui_timeout_id) == 1) {
+                close_all = true;
+                gui_timer_delete(gui_timeout_id);
```

Comment:
I do not like this logic, creating gui_timer in gui.cpp and deleting it here
Also GuiDefaults::menu_timeout_enabled is redundand, because it was already checked in gui.cpp while creating gui_timer

In this case I would not use gui_timer. Would be better readable to handle entire logic inside this module.